### DISCUSSION TITLE
NUCLEO-H723 adc on port A

### DISCUSSION THREAD

**Discussion by brave-ulysses** [micropython/micropython#11341] [STM32 / Pyboard]
anyone had success using PA0/3 for adc input? the inputs on PC0/3 work as expected

using MicroPython v1.19.1-992-g38e7b842c-dirty on 2023-04-25; FJ_H723 with STM32H723ZGT6

```
def par(r1,r2):
    return 1/(1/(r1) + 1/(r2))

def vdiv(r1,r2):
    return((r1)/((r1)+(r2)))

MON_12v=machine.ADC(machine.Pin.cpu.A3)
MON_m5v=machine.ADC(machine.Pin.cpu.C0)
MON_5v=machine.ADC(machine.Pin.cpu.A0)
MON_m48v=machine.ADC(machine.Pin.cpu.C3)
adcint = pyb.ADCAll(12, 0x70000)

mon_3v3  = adcint.read_vref()
mon_12v_r  = MON_12v.read_u16()/2**16*mon_3v3
mon_m5v_r  = MON_m5v.read_u16()/2**16*mon_3v3
mon_5v_r   = MON_5v.read_u16()/2**16*mon_3v3
mon_m48v_r = MON_m48v.read_u16()/2**16*mon_3v3

mon_12v  = mon_12v_r / vdiv(3.3e3,15e3)
mon_m5v  = mon_m5v_r / vdiv(par(15e3,200e3),15e3)
mon_5v   = mon_5v_r / vdiv(15e3,15e3)
mon_m48v = mon_m48v_r / vdiv( par(10e3,200e3),200e3)
```

verified 2.5v on PA0;  read_u16 returns ~1400

af pin functions. looks like any one of the 3 adc units can be mapped to these pins
grep P[AC][03] stm32h723_af.csv
PortA,PA0,,TIM2_CH1/TIM2_ETR,TIM5_CH1,TIM8_ETR,TIM15_BKIN,SPI6_NSS/I2S6_WS,,USART2_CTS/USART2_NSS,UART4_TX,SDMMC2_CMD,SAI4_SD_B,ETH_MII_CRS,FMC_A19,,,EVENTOUT,ADC123_IN0
PortA,PA3,,TIM2_CH4,TIM5_CH4,LPTIM5_OUT,TIM15_CH2,I2S6_MCK,OCTOSPIM_P1_IO2,USART2_RX,,LCD_B2,OTG_HS_ULPI_D0,ETH_MII_COL,OCTOSPIM_P1_CLK,,LCD_B5,EVENTOUT,ADC123_IN3
PortC,PC0,,FMC_D12/FMC_AD12,,DFSDM1_CKIN0,,,DFSDM1_DATIN4,,SAI4_FS_B,FMC_A25,OTG_HS_ULPI_STP,LCD_G2,FMC_SDNWE,,LCD_R5,EVENTOUT,ADC123_IN10
PortC,PC3,PWR_SLEEP,,,DFSDM1_DATIN1,OCTOSPIM_P1_IO6,SPI2_MOSI/I2S2_SDO,,,,OCTOSPIM_P1_IO0,OTG_HS_ULPI_NXT,ETH_MII_TX_CLK,FMC_SDCKE0,,,EVENTOUT,ADC123_IN13

**Comment by brave-ulysses** [comment]
PA0, PA3 adc pin definitions in csv file don't match stm doc

https://www.st.com/resource/en/datasheet/stm32h723zg.pdf

![image](https://user-images.githubusercontent.com/22964897/235217957-c4402c28-e984-49fd-bea8-b192b7c2bcf4.png)

**Comment by brave-ulysses** [comment]
submitted:
boards: NUCLEO_H723ZG: ADC AF definitions. #11373
@pschindler
@softcity003
@PYTRONIX-INC

**Reply by pschindler** [reply]
Thank you very much for this. I have observed this issue, but did not have time to look into this.
